test/plots/stackedBarPlotTests.ts

```diff
@@ -1,607 +1,411 @@
-      plot.x((d) => d.x, xScale);
-      plot.y((d) => d.y, yScale);
-      let xAxis = new Plottable.Axes.Category(xScale, "bottom");
-      new Plottable.Components.Table([[plot], [xAxis]]).renderTo(svg);
+      it("draws bars at specified x location and stacks in order of datasets", () => {
+        let bars = stackedBarPlot.content().selectAll("rect");
+
+        let datumCount = stackedBarPlot.datasets().reduce((prev, curr) => prev + curr.data().length, 0);
+        assert.strictEqual(bars.size(), datumCount, "draws a bar for each datum");
+
+        let domain = xScale.domain();
+        domain.forEach((value) => {
+          let domainBarPairs = d3.pairs(bars.filter((d) => d.x === value)[0]);
+          domainBarPairs.forEach((aBarPair) => {
+            assert.closeTo(TestMethods.numAttr(d3.select(aBarPair[0]), "x"), TestMethods.numAttr(d3.select(aBarPair[1]), "x"),
+              window.Pixel_CloseTo_Requirement, "bars at same x position");
+            assert.operator(TestMethods.numAttr(d3.select(aBarPair[0]), "y"), ">",
+              TestMethods.numAttr(d3.select(aBarPair[1]), "y"), "previous dataset bar under second");
+          });
+        });
+        stackedBarPlot.destroy();
+        svg.remove();
+      });
     });
 
-    it("renders correctly", () => {
-      let bars = (<any> plot)._renderArea.selectAll("rect");
-
-      assert.lengthOf(bars[0], 7, "draws a bar for each datum");
-
-      let aBars = [d3.select(bars[0][0]), d3.select(bars[0][3])];
-
-      let bBars = [d3.select(bars[0][1]), d3.select(bars[0][4]), d3.select(bars[0][5])];
-
-      let cBars = [d3.select(bars[0][2]), d3.select(bars[0][6])];
-
-      assert.closeTo(TestMethods.numAttr(aBars[0], "x"), TestMethods.numAttr(aBars[1], "x"), 0.01, "A bars at same x position");
-      assert.operator(TestMethods.numAttr(aBars[0], "y"), ">", TestMethods.numAttr(aBars[1], "y"), "first dataset A bar under second");
-
-      assert.closeTo(TestMethods.numAttr(bBars[0], "x"), TestMethods.numAttr(bBars[1], "x"), 0.01, "B bars at same x position");
-      assert.closeTo(TestMethods.numAttr(bBars[1], "x"), TestMethods.numAttr(bBars[2], "x"), 0.01, "B bars at same x position");
-      assert.operator(TestMethods.numAttr(bBars[0], "y"), ">", TestMethods.numAttr(bBars[1], "y"), "first dataset B bar under second");
-      assert.operator(TestMethods.numAttr(bBars[1], "y"), ">", TestMethods.numAttr(bBars[2], "y"), "second dataset B bar under third");
-
-      assert.closeTo(TestMethods.numAttr(cBars[0], "x"), TestMethods.numAttr(cBars[1], "x"), 0.01, "C bars at same x position");
-      assert.operator(TestMethods.numAttr(cBars[0], "y"), ">", TestMethods.numAttr(cBars[1], "y"), "first dataset C bar under second");
-
-      svg.remove();
+    describe("fail safe tests", () => {
+
+      it("should default to 0 when calculating stack offsets with non-numbers", () => {
+        let svg = TestMethods.generateSVG();
+        let data1 = [
+          { x: "A", y: "s"},
```

Comment:
well, technically, `NaN` is `number` as well
just saying those are edge cases to be tested against